lib/trie/database.go

```diff
@@ -179,6 +185,21 @@ func (t *Trie) load(db chaindb.Database, curr Node) error {
 		}
 	}
 
+	for _, key := range t.GetKeysWithPrefix(ChildStorageKeyPrefix) {
+		childTrie := NewEmptyTrie()
+		value := t.Get(key)
+		err := childTrie.Load(db, common.NewHash(value))
+		if err != nil {
+			return fmt.Errorf("failed to load child trie with root hash=0x%x: %w", value, err)
+		}
+
+		err = t.PutChild(value, childTrie)
+		if err != nil {
+			return fmt.Errorf("failed to insert child trie with root hash=0x%x into main trie: %w",
+				childTrie.root.GetHash(), err)
+		}
+	}
```

Comment:
nit could we have this in its own method like `loadChildTries` and call it here?

Reply:
generally you split when you want to reuse the code or make a large piece of code more readable. This function is not that big. Separating this logic, would not achieve much.

Reply:
Yes but also it's easier to test smaller bits. I don't think there is a test covering this specifically for example right (there might a test for a function calling that one though)?

Reply:
Spent some time on this and wasn't obvious how to test this. I have created a separate issue for this and assigned it to myself. https://github.com/ChainSafe/gossamer/issues/2171